full-stack-practice/client/src/Catalog.tsx

```diff
@@ -0,0 +1,76 @@
+import { useEffect, useState } from 'react';
+import { Product } from '../../server/server.ts';
+import { toDollars } from '../src/lib/index';
+import { Link } from 'react-router-dom';
+import { ClientError } from '../../server/lib/client-error.ts';
+export function Catalog() {
+  const [products, setProducts] = useState<Product[]>([]);
+  const [isLoading, setIsLoading] = useState(true);
+  const [error, setError] = useState<unknown>();
+
+  useEffect(() => {
+    async function loadItems() {
+      try {
+        const response = await fetch('/api/products');
+        if (!response.ok) {
+          throw new ClientError(400, 'result is invalid');
+        }
+        const items = await response.json();
+        if (!items) {
```

Comment:
Presumably this is not necessary because the server should return a 400 or 500 error if it cannot return correct items. And in that case, you would throw an exception on line 16 so you would never execute this line of code.